src/rules/no_default_default.rs

```diff
@@ -1,42 +1,86 @@
-use crate::{rule, rule::Rule, rule_listener, ViolationBuilder};
+use std::sync::Arc;
 
-pub fn no_default_default_rule() -> Rule {
-    rule! {
-        name => "no_default_default",
-        fixable => true,
-        create => |_context| {
-            vec![
-                rule_listener! {
-                    query => r#"(
-                      (call_expression
-                        function:
-                          (scoped_identifier
-                            path:
-                              (identifier) @first (#eq? @first "Default")
-                            name:
-                              (identifier) @second (#eq? @second "default")
-                          )
-                      ) @c
-                    )"#,
-                    capture_name => "c",
-                    on_query_match => |node, query_match_context| {
-                        query_match_context.report(
-                            ViolationBuilder::default()
-                                .message(r#"Use '_d()' instead of 'Default::default()'"#)
-                                .node(node)
-                                .fix(|fixer| {
-                                    fixer.replace_text(node, "_d()");
-                                })
-                                .build()
-                                .unwrap(),
-                        );
-                    }
-                }
-            ]
+use tree_sitter::Node;
+use tree_sitter_grep::SupportedLanguage;
+
+use crate::{
+    context::QueryMatchContext,
+    rule::{FileRunInfo, Rule, RuleInstance, RuleInstancePerFile, RuleListenerQuery, RuleMeta},
+    Config, ViolationBuilder,
+};
+
+pub struct NoDefaultDefaultRule {
+    listener_queries: Vec<RuleListenerQuery>,
+}
+
+impl Rule for NoDefaultDefaultRule {
+    fn meta(&self) -> RuleMeta {
+        RuleMeta {
+            name: "no_default_default".to_owned(),
+            fixable: true,
+            languages: vec![SupportedLanguage::Rust],
```

Comment:
I don't think I'm doing anything with this yet but this was a step toward "rules know which language(s) they are for"